src/internal/m365/collection/drive/collections.go

```diff
@@ -714,6 +722,95 @@ func (c *Collections) getCollectionPath(
 	return collectionPath, nil
 }
 
+type driveEnumerationStats struct {
+	numPages      int
+	numAddedFiles int
+	numContainers int
+	numBytes      int64
+}
+
+func newPagerLimiter(opts control.Options) *pagerLimiter {
+	res := &pagerLimiter{
+		isPreview:            opts.ToggleFeatures.PreviewBackup,
+		maxContainers:        opts.ItemLimits.MaxContainers,
+		maxItemsPerContainer: opts.ItemLimits.MaxItemsPerContainer,
+		maxItems:             opts.ItemLimits.MaxItems,
+		maxBytes:             opts.ItemLimits.MaxBytes,
+		maxPages:             opts.ItemLimits.MaxPages,
+	}
+
+	if res.maxContainers == 0 {
+		res.maxContainers = defaultPreviewNumContainers
+	}
+
+	if res.maxItemsPerContainer == 0 {
+		res.maxItemsPerContainer = defaultPreviewNumItemsPerContainer
+	}
+
+	if res.maxItems == 0 {
+		res.maxItems = defaultPreviewNumItems
+	}
+
+	if res.maxBytes == 0 {
+		res.maxBytes = defaultPreviewNumBytes
+	}
+
+	if res.maxPages == 0 {
+		res.maxPages = defaultPreviewNumPages
+	}
+
+	return res
+}
+
+type pagerLimiter struct {
+	isPreview            bool
```

Comment:
since `isPreview` is already implicitly equal to `enabled`.  
```suggestion
	enabled            bool
```

Reply:
going to play with the name a bit since there's also the `enabled` function